### DISCUSSION TITLE
Folders per language <title>

### DISCUSSION THREAD

**Discussion by RuizJpaul** [community/community#43543] [Programming Help]
### Little programs  

Question

### Body

I'm a begginer so maybe i'm mistaken; my question beggins on how could i upload my projects in an optimized way, i'm just programming "tiny" codes and to upload a project it's a new repository per project (it's too much space for so little programs).
If there is a way to sort many little projects inside one folder according its language, it would be great for me to you let me know.
Thanks anyways.

**Comment by jfullstackdev** [comment]
simply do that, there is no harm doing that, create every folder for each file with another language

**Comment by auchynnikau** [comment]
Git Submodules might be helpful in your case, you can read more about it [here](https://www.atlassian.com/git/tutorials/git-submodule) and [here](https://git-scm.com/book/en/v2/Git-Tools-Submodules).

But the easiest way is to simply create folder for each specific language within your repository. It's totally fine solution until your programs become too big. Also make sure that your root `.gitignore` file contains rules for all languages you're using or it will be even better approach to have a multiple `.gitignore` files for each subfolder not to mix all rules together.

**Comment by sergeyleschev** [comment]
One way to optimize your project uploads is to create a separate folder for each language within your repository. This will help to organize and group your projects in a structured and easy-to-find manner. Additionally, you can make use of Git Submodules to link different repositories together.
